src/MilestoneProcessor.ts

```diff
@@ -1,74 +1,51 @@
 import * as core from '@actions/core';
 import * as github from '@actions/github';
-import {Octokit} from '@octokit/rest';
+import { Octokit } from '@octokit/rest';
+import { Context } from '@actions/github/lib/context';
+import { WebhookPayload } from '@actions/github/lib/interfaces';
+import { MilestoneProcessorOptions, Issue, Milestone, ActionEvent } from './interfaces';
 
-type OctoKitIssueList = Octokit.Response<Octokit.IssuesListForRepoResponse>;
-type OctoKitMilestoneList = Octokit.Response<
-  Octokit.IssuesListMilestonesForRepoResponse
->;
+type OctoKitPullsList = Octokit.Response<Octokit.PullsListResponse>;
+type OctoKitIssuesList = Octokit.Response<Octokit.IssuesListForRepoResponse>;
+type OctoKitMilestoneList = Octokit.Response<Octokit.IssuesListMilestonesForRepoResponse>;
+type OctoKitCommitsPullsList = Octokit.Response<Octokit.ReposListPullRequestsAssociatedWithCommitResponse>;
 
 const OPERATIONS_PER_RUN = 100;
-// TODO: Expose as option.
-const MIN_ISSUES_IN_MILESTONE = 3;
-
-export interface Issue {
-  title: string;
-  number: number;
-  updated_at: string;
-  labels: Label[];
-  pull_request: any;
-  state: string;
-  locked: boolean;
-}
-
-export interface Milestone {
-  id: number;
-  title: string;
-  number: number;
-  updated_at: string;
-  description: string;
-  open_issues: number;
-  closed_issues: number;
-  state: string;
-}
-
-export interface Label {
-  name: string;
-}
-
-export interface MilestoneProcessorOptions {
-  repoToken: string;
-  debugOnly: boolean;
-}
 
 /***
  * Handle processing of issues for staleness/closure.
  */
 export class MilestoneProcessor {
   readonly client: github.GitHub;
   readonly options: MilestoneProcessorOptions;
-  private operationsLeft: number = 0;
-
   readonly staleIssues: Issue[] = [];
   readonly closedIssues: Issue[] = [];
   readonly closedMilestones: Milestone[] = [];
+  readonly closedEvents: ActionEvent[] = [];
+
+  private operationsLeft: number = 0;
+  private relatedNotFound: boolean = false;
+  private detectedEvent: string | undefined = "none";
+  private eventPullRequest: boolean = false;
+  private eventPush: boolean = false;
 
   constructor(
     options: MilestoneProcessorOptions,
-    getMilestones?: (page: number) => Promise<Milestone[]>
+    getMilestones?: (page: number) => Promise<Milestone[]>,
   ) {
     this.options = options;
     this.operationsLeft = OPERATIONS_PER_RUN;
     this.client = new github.GitHub(options.repoToken);
+    this.detectedEvent = process.env.GITHUB_EVENT_NAME;
+    this.eventPullRequest = (this.options.debugOnly ? this.options.relatedOnly : this.getCheckPullRequest(github.context));
```

Comment:
Needed for issues too?

Reply:
hmm actually the `detectedEvent` should be removed, forgot to remove it from here earlier, and the `eventPullRequest` is just to check if the event/action is pr